lib/Sema/TypeCheckDecl.cpp

```diff
@@ -7950,12 +7950,11 @@ void TypeChecker::synthesizeMemberForLookup(NominalTypeDecl *target,
     }
   } else {
     auto argumentNames = member.getArgumentNames();
-    if (argumentNames.size() != 1)
+    if (member.isCompoundName() && argumentNames.size() != 1)
       return;
 
-    auto argumentName = argumentNames.front();
     if (baseName == DeclBaseName::createConstructor() &&
-        argumentName == Context.Id_from) {
+        (member.isSimpleName() || argumentNames.front() == Context.Id_from)) {
```

Comment:
If `member` is a compound name, won't we fall into here regardless of what the argument name is?

Reply:
Unfortunately I haven't kept up with the `isSpecialName` changes, so correct me if I'm wrong, but if `member.isSimpleName()`, what makes it okay to ignore `argumentNames`? Wouldn't we end up falling into this case unnecessarily even when the argument name isn't `from:`? (That's what I'm trying to get at here.)

Reply:
A simple name is like `encode` or `init`. A compound name is `encode(to:)` or `init(from:)`. So this code was doing the right synthesis whenever someone asked for `init(from:)` explicitly, but not if they just asked for `init`. The latter is what happens when trying to inherit initializers.

Reply:
Got it, thanks for the clarification! Out of curiosity, then, is this conformance check not going to be expensive to perform on _every_ inheritance of an `init`? Why does inheritance not specify the full compound name?

Reply:
This is trying to figure out what initializers there are at all, so the only thing it can look up is `init`. Indeed, because we can synthesize `init(from:)`, any subclassing action has to do this check.

Reply:
Okay, I guess there isn't much we can do about that here without changing the inheritance mechanism.

My last lingering question is where this inheritance check might happen relative to regular protocol conformance checking; is it possible for this to trigger a conformance check on a subclass before the superclass is ever checked? Would that correctly trigger a check on the superclass first?

Reply:
Excellent — thanks for fielding my concerns! Looks good once we figure out the source of these failures.